src/vs/workbench/contrib/terminalContrib/chat/browser/terminalChatWidget.ts

```diff
@@ -161,147 +138,13 @@ export class TerminalChatWidget extends Disposable {
-				alwaysConsumeMouseWheel: false
-			},
-			lineDecorationsWidth: 0,
-			overviewRulerBorder: false,
-			scrollBeyondLastLine: false,
-			renderLineHighlight: 'none',
-			fixedOverflowWidgets: true,
-			dragAndDrop: false,
-			revealHorizontalRightPadding: 5,
-			minimap: { enabled: false },
-			guides: { indentation: false },
-			rulers: [],
-			renderWhitespace: 'none',
-			dropIntoEditor: { enabled: true },
-			quickSuggestions: false,
-			suggest: {
-				showIcons: false,
-				showSnippets: false,
-				showWords: true,
-				showStatusBar: false,
-			},
-			wordWrap: 'on'
-		}, { isSimpleWidget: true }));
-		this._editor = editor;
-		this._register(editor.onDidFocusEditorText(() => this._responseEditorFocusedContextKey.set(true)));
-		this._register(editor.onDidBlurEditorText(() => this._responseEditorFocusedContextKey.set(false)));
-		this._register(Event.any(editor.onDidChangeModelContent, editor.onDidChangeModelDecorations)(() => {
-			const height = editor.getContentHeight();
-			editor.layout(new Dimension(640, height));
-		}));
-
-		this.model = this._getTextModel(URI.from({
-			path: `terminal-inline-chat-${this._instance.instanceId}`,
-			scheme: 'terminal-inline-chat',
-			fragment: initialCommandResponse
-		}));
-		this.model.then(model => {
-			if (model) {
-				// Initial layout
-				editor.layout(new Dimension(640, 0));
-				editor.setModel(model);
-				const height = editor.getContentHeight();
-				editor.layout(new Dimension(640, height));
-
-				// Initialize language
-				const languageId = this._getLanguageFromShell(shellType);
-				model.setLanguage(languageId);
-			}
-		});
-	}
-
-	private _getAriaLabel(): string {
-		const verbose = this._configurationService.getValue<boolean>(AccessibilityVerbositySettingId.Chat);
-		if (verbose) {
-			// TODO: Add verbose description
-		}
-		return localize('terminalResponseEditor', "Terminal Response Editor");
-	}
-
-	private async _getTextModel(resource: URI): Promise<ITextModel | null> {
-		const existing = this._modelService.getModel(resource);
-		if (existing && !existing.isDisposed()) {
-			return existing;
-		}
-		return this._modelService.createModel(resource.fragment, null, resource, false);
-	}
-
-	private _getLanguageFromShell(shell?: string): string {
-		switch (shell) {
-			case 'fish':
-				return this._languageService.isRegisteredLanguageId('fish') ? 'fish' : 'shellscript';
-			case 'zsh':
-				return this._languageService.isRegisteredLanguageId('zsh') ? 'zsh' : 'shellscript';
-			case 'bash':
-				return this._languageService.isRegisteredLanguageId('bash') ? 'bash' : 'shellscript';
-			case 'sh':
-				return 'shellscript';
-			case 'pwsh':
-				return 'powershell';
-			default:
-				return 'plaintext';
-		}
-	}
-
-	setValue(value: string) {
-		this._editor.setValue(value);
-	}
+function parseCodeFromBlock(block?: string): string | undefined {
+	const match = block?.match(/```.*?\n([\s\S]*?)```/);
+	return match ? match[1].trim() : undefined;
+}
```

Comment:
Can we use the implementation of `workbench.action.chat.copyCodeBlock` for fetching the code block?

Reply:
I looked into that. Since we are not using a `IChatWidget`, this is undefined 

https://github.com/microsoft/vscode/blob/6c6bcf374da88ce4c43fb9165c5707b88ef16645/src/vs/workbench/contrib/chat/browser/actions/chatCodeblockActions.ts#L578-L582

Reply:
inline chat widget uses the `ChatListItemRenderer` that keeps track of those code blocks, so maybe I can expose that on the inline chat widget

Reply:
I think the context is meant to already be there thanks to this?

https://github.com/microsoft/vscode/blob/38f6b07e3141a07eb3d2d8487ef8af0a74165f03/src/vs/workbench/contrib/chat/browser/codeBlockPart.ts#L375-L389

Provided inline chat uses this same code block code of the chat view

Reply:
yeah, I think `InlineChatWidget` would need similar functions to those of the `ChatWidget`

https://github.com/microsoft/vscode/blob/673b05c784f0f5ca6b3e161b8acd040c76d40c78/src/vs/workbench/contrib/chat/browser/chatWidget.ts#L649-L655

Reply:
ok, spent a bunch of time on this. because we don't render in the chat view, `IChatResponseViewModel` does not exist for us. so we cannot do something similar here

Reply:
currently, we don't have a uri here for ours. I think we need this and then can call `getCodeBlockInfoForEditor`
https://github.com/microsoft/vscode/blob/673b05c784f0f5ca6b3e161b8acd040c76d40c78/src/vs/workbench/contrib/chat/browser/chatListRenderer.ts#L908-L912

Reply:
I think the issue is we don't use a `ChatViewModel`, just a `ChatModel`. https://github.com/microsoft/vscode/blob/673b05c784f0f5ca6b3e161b8acd040c76d40c78/src/vs/workbench/contrib/terminalContrib/chat/browser/terminalChatController.ts#L85

Reply:
@Tyriar the issue is that this code does nothing for us. 

https://github.com/microsoft/vscode/blob/673b05c784f0f5ca6b3e161b8acd040c76d40c78/src/vs/workbench/contrib/chat/browser/actions/chatCodeblockActions.ts#L577-L585

Reply:
I figured out how the chat code actions actually get the code - they just use the active editor's value, so that's what we'll do.

Reply:
because we have to rely on the context of the focused editor, we'll do away with our button actions in favor of toolbar ones.